### PR TITLE
Onnx support: add a dim parameter to ops.Gather

### PR DESCRIPTION
**Pull request by lingxiao1989**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this patch)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Gather.scala [hunk 1]
```diff
@@ -61,16 +69,15 @@ class Gather[T: ClassTag, D: ClassTag]()(implicit ev: TensorNumeric[T], ev2: Ten
       var i = 0
       while (i < indices.nElement()) {
         val index = indices.valueAt(i + 1)
-        require(index < inputSizes(0),
-          s"index should smaller than ${inputSizes(0)}, but got $index")
-        output.select(1, i + 1).copy(inputTensor.select(1, index + 1))
+        require(index < inputSizes(dim - 1),
+          s"index should smaller than ${inputSizes(dim - 1)}, but got $index")
+        output.select(dim, i + 1).copy(inputTensor.select(dim, index + 1))
```
Comment (LeicongLi) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Gather.scala]: line#58-line#62 and line#72-line#74 are very similar, I would suggest try to simplify and make them reusable, what do you think?

**Comment by lingxiao1989** [comment]
Jenkins:
http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/928/

**Comment by LeicongLi** [comment]
Looks good to me.

**Comment by lingxiao1989** [comment]
http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/952/

**Pull request merged**